hs_core/serialization.py

```
@@ -0,0 +1,1061 @@
+    is a subclass of hs_core.model.BaseResource representing the newly created
+    resource. Once the dependant resource has been created, the caller is
+    responsible for calling ResourceMeta.write_metadata_to_resource(Resource)
+    to ensure that all metadata fields have been populated.
+
+    :raises: HsDeserializationException if an error occurred.
+    """
+    # Get resource metadata
+    resource_files = None
+    rm = None
+    try:
+        rm = GenericResourceMeta.read_metadata_from_resource_bag(bag_content_path)
+        # Filter resource files
+        resource_files_filtered = []
+        for f in rm.files:
+            if rm.include_resource_file(f):
+                resource_files_filtered.append(f)
+        res_files = [os.path.join(bag_content_path, f) for f in resource_files_filtered]
+        resource_files = _prepare_resource_files_for_creation(res_files)
+
+    except GenericResourceMeta.ResourceMetaException as e:
+        msg = "Error occurred while trying to create resource from bag path {0}. "
+        msg += "Error was: {1}"
+        msg = msg.format(bag_content_path, str(e))
+        raise HsDeserializationException(msg)
+
+    # Send pre-create resource signal
+    try:
+        if resource_files is None:
+            resource_files = []
+
+        page_url_dict, res_title, metadata = resource_pre_create_actions(resource_type=rm.res_type,
+                                                                         files=resource_files,
+                                                                         resource_title=rm.title,
+                                                                         page_redirect_url_key=None)
+    except ResourceFileSizeException as ex:
+        raise HsDeserializationException(ex.message)
+
+    except ResourceFileValidationException as ex:
+        raise HsDeserializationException(ex.message)
+
+    except Exception as ex:
+        raise HsDeserializationException(ex.message)
```

This could really mask errors that happen underneath, making bugs harder to track down.